anaconda: Use flattened RPM Packages directory when using CD/DVD media

This is a backport of an Anaconda 17 patch.

Signed-off by: Rob Woolley <rob.woolley@windriver.com>

commit 6475cfd6abd6ff86c8fad4198fbddf1dfb9f7f91
Author: Will Woods <wwoods@redhat.com>
Date:   Fri May 4 15:41:21 2012 -0400

    Fix anaconda not finding CD when booting without root=live:<CD>
    
    Basically the same deal as commit 45363385 - look at /run/install/repo
    (where the CD will be mounted) when looking for media.
    
    Obviously, scanForMedia should handle already-mounted media instead, but
    that's something for F18.

Index: b/pyanaconda/yuminstall.py
===================================================================
--- a/pyanaconda/yuminstall.py
+++ b/pyanaconda/yuminstall.py
@@ -493,6 +493,11 @@ class AnacondaYum(yum.YumBase):
             isys.mount("/run/initramfs/live/", self.tree, bindMount=True)
             self.mediagrabber = self.mediaHandler
             self._baseRepoURL = "file://%s" % self.tree
+        elif os.path.isdir("/run/install/repo/repodata"):
+            # Same hack as above. FIXME: make scanForMedia do this, dammit
+            isys.mount("/run/install/repo", self.tree, bindMount=True)
+            self.mediagrabber = self.mediaHandler
+            self._baseRepoURL = "file://%s" % self.tree
         else:
             cdr = scanForMedia(self.tree, self.anaconda.storage)
             if cdr:
